src/Components/CheckBox/CheckBox.tsx

```diff
@@ -0,0 +1,36 @@
+import React from "react";
+import { CheckBoxType } from "./consts";
+import { Checkbox, FormControlLabel } from "@mui/material";
+import { CustomCheckbox } from "./styles";
+import IndeterminateCheckBoxOutlinedIcon from "@mui/icons-material/IndeterminateCheckBoxOutlined";
+
+export type CheckBoxProps = {
+  type?: CheckBoxType;
+  onChange?: (event: React.ChangeEvent<HTMLInputElement>) => void;
+  label?: string;
+  disabled?: boolean;
+};
```

Comment:
1. I would change the `type` ,`onChange` props to be required and not optional (without the `?`).
you need to demand the user to pass those props since they are essential for the checkbox behaviour.
And if for instance you dont want the checkbox to do nothing, you can pass `onChange={() =>}`.
2. Pay attention that you have duplication in the disabled behaviour: you pass it as a prop, but it is also an option in the `CheckBoxType`. I would remove it from the `CheckBoxType`.